File: Ravi-Teja/Subsets.cpp
class Solution {
public:
    vector<vector<int>> ans;
    void dfs(int n, vector<int> &temp, vector<int> &nums){
        if(n<0){
            ans.push_back(temp);
            return;
        }
        temp.push_back(nums[n]);
        dfs(n-1, temp, nums);
        temp.pop_back();
        dfs(n-1, temp, nums);
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        ans.clear();
        vector<int> temp;
        dfs(nums.size()-1, temp, nums);
        return ans;
    }
};

File: Parth Mathur/Subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        ans=new ArrayList<>();
        sets(nums,0,new ArrayList<Integer>());
        return ans;
    }
    List<List<Integer>> ans;
    void  sets(int[] nums,int idx,List<Integer> temp){
        if(idx==nums.length){
            ans.add(new ArrayList<>(temp));
            return;
        }
        ans.add(new ArrayList<>(temp));
        for(int i=idx;i<nums.length;i++){
            temp.add(nums[i]);
            sets(nums,i+1,temp);
            temp.remove(temp.size()-1);
        }
    }
}
